### ISSUE TITLE
[build] `rcore` module does not compile under linux with `-std=c99` *without* `-D_GNU_SOURCE`

### ISSUE THREAD

**Issue by Peter0x44** [raysan5/raylib#2729]
Please, before submitting a new issue verify and check:

 - [ ] I tested it on latest raylib version from master branch
 - [ ] I checked there is no similar issue already reported

### Issue description
Under Linux, rcore.c cannot compile as c99, without defining _GNU_SOURCE

`gcc -c rcore.c -DGRAPHICS_API_OPENGL_33 -DPLATFORM_DESKTOP -std=c99`
```
In file included from rcore.c:128:
rgestures.h: In function ‘rgGetCurrentTime’:
rgestures.h:540:21: error: storage size of ‘now’ isn’t known
  540 |     struct timespec now;
      |                     ^~~
rgestures.h:541:5: warning: implicit declaration of function ‘clock_gettime’ [-Wimplicit-function-declaration]
  541 |     clock_gettime(CLOCK_MONOTONIC, &now);
      |     ^~~~~~~~~~~~~
rgestures.h:541:19: error: ‘CLOCK_MONOTONIC’ undeclared (first use in this function)
  541 |     clock_gettime(CLOCK_MONOTONIC, &now);
      |                   ^~~~~~~~~~~~~~~
rgestures.h:541:19: note: each undeclared identifier is reported only once for each function it appears in
rcore.c: In function ‘GetApplicationDirectory’:
rcore.c:3074:19: warning: implicit declaration of function ‘readlink’ [-Wimplicit-function-declaration]
 3074 |     ssize_t len = readlink("/proc/self/exe", appDir, size);
      |                   ^~~~~~~~
rcore.c: In function ‘WaitTime’:
rcore.c:4782:16: error: variable ‘req’ has initializer but incomplete type
 4782 |         struct timespec req = { 0 };
      |                ^~~~~~~~
rcore.c:4782:33: warning: excess elements in struct initializer
 4782 |         struct timespec req = { 0 };
      |                                 ^
rcore.c:4782:33: note: (near initialization for ‘req’)
rcore.c:4782:25: error: storage size of ‘req’ isn’t known
 4782 |         struct timespec req = { 0 };
      |                         ^~~
rcore.c:4789:16: warning: implicit declaration of function ‘nanosleep’ [-Wimplicit-function-declaration]
 4789 |         while (nanosleep(&req, &req) == -1) continue;
      |                ^~~~~~~~~
```
 Now, adding -D_GNU_SOURCE to that command, it compiles without any errors or warnings

There appears to be some code already trying to handle this case:
https://github.com/raysan5/raylib/blob/587e61def9450dcf64648177e1158acc9762f2a2/src/rcore.c#L154-L157
But, it doesn't seem to work

The raylib Makefile adds -D_GNU_SOURCE, which could be a reason this is overlooked for so long

It also compiles with `-std=gnu99`, which seems to be what is allowing CMake to work, it uses it by default.

This issue can be reproduced with CMake too, if it is invoked like this:
`cmake .. -DCMAKE_C_EXTENSIONS=OFF`

**Comment by raysan5** [comment]
@Peter0x44 Could this define just be added under the mentioned lines in `rcore`? (I don't know it it could have any side effect)

**Comment by Peter0x44** [comment]
@raysan5 I think it can, but perhaps not in that exact place in rcore
One thing to be aware of is that it must be defined *before* any other header is included, because some headers can apparently include each other

`man feature_test_macros` for me says:
```
       NOTE:  In order to be effective, a feature test macro must be defined before including any header files.  This can be done either in the compilation command (cc -DMACRO=value) or by defining the macro within the source code before including any headers.  The requirement that the macro must be defined
       before including any header file exists because header files may freely include one another.  Thus, for example, in the following lines, defining the _GNU_SOURCE macro may have no effect because the header <abc.h> itself includes <xyz.h> (POSIX explicitly allows this):
````

**Comment by raysan5** [comment]
@Peter0x44 I reviewed this issue, as far as it is a compilation flag, shouldn't it be added to the build system?
